imap_processing/spice/geometry.py

```diff
@@ -644,3 +684,32 @@ def spherical_to_cartesian(spherical_coords: NDArray, degrees: bool = False) ->
     cartesian_coords = np.stack((x, y, z), axis=-1)
 
     return cartesian_coords
+
+
+def get_right_ascension_and_declination(coords: npt.NDArray) -> npt.NDArray:
+    """
+    Convert rectangular coordinates to range, right ascension, and declination.
+
+    This is a vectorized wrapper around `spiceypy.recrad`
+    https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/cspice/recrad_c.html
+
+    Parameters
+    ----------
+    coords : np.ndarray
+        Cartesian (rectangular) coordinates with shape `(3)` or `(n, 3)`,
```

Comment:
Similar to my comment for the `rotation_matrix_to_euler_angles` function, I wonder if it would be a good idea to add error handling around the number of dimensions that coords is passed in with? Like, if it's not 1d or 2d, it returns an error?